Sources/AEPOptimize/OptimizeError.swift

```diff
@@ -0,0 +1,45 @@
+// Delete this line
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import Foundation
+
+/// AEPOptimizeError class used to create AEPOptimizeError from error details received from Experience Edge.
+@objc(AEPOptimizeError)
+public class AEPOptimizeError: NSObject {
+    public let type: String?
+    public let status: Int?
+    public let title: String?
+    public let detail: String?
+    public var aepError = AEPError.unexpected
```

Comment:
Wondering why we have `AEPError` here? Could you please share how customers are going to use this?

Reply:
In get proposition api, today we return ([DecisionScope: OptimizeProposition]?, Error?), where Error is of type AEPError. Considering the possibility customer has kept some check or performed any logic on their side on basis of AEPErrors we added this variable. We can discuss on sending back AEPOptimizeError in getProposition api as well instead of AEPError to improve on parity between api callback.